enterprise/reporting/error_extractor.go

```diff
@@ -0,0 +1,257 @@
+package reporting
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/samber/lo"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	responseKey = "response"
+	spaceStr    = " "
+
+	destinationResponseKey = "destinationResponse"
+	errorsKey              = "errors"
+)
+
+var (
+	urlRegex         = regexp.MustCompile(`\b((?:https?://|www\.)\S+)\b`)
+	ipRegex          = regexp.MustCompile(`\b(?:\d{1,3}\.){3}\d{1,3}\b`)
+	emailRegex       = regexp.MustCompile(`\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b`)
+	notWordRegex     = regexp.MustCompile(`\W+`)
+	idRegex          = regexp.MustCompile(`\b([a-zA-Z0-9-]*\d[a-zA-Z0-9-]*)\b`)
+	spaceRegex       = regexp.MustCompile(`\s+`)
+	whitespacesRegex = regexp.MustCompile("[ \t\n\r]*") // used in checking if string is a valid json to remove extra-spaces
+
+	defaultErrorMessageKeys = []string{"message", "description", "detail", "title", "error", "error_message"}
+)
+
+type ExtractorT struct {
```

Comment:
Beware that we are trying to get rid of the `TypeT` and `InterfaceI` types since they are not go idiomatic nor conventional. Whenever possible please try to name your types without the suffix.

Reply:
Removed them from this PR changes.